velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -217,7 +218,20 @@ dwio::common::StripeProgress getStripeProgress(
  *
  * This method assumes each input `ColumnarBatch` have same schema.
  */
-void Writer::write(const VectorPtr& data) {
+void Writer::write(const VectorPtr& dataToWrite) {
+  VectorPtr data = dataToWrite;
+  if (data && schema_) {
+    auto rowType = std::dynamic_pointer_cast<const velox::RowType>(schema_);
```

Comment:
Can we require schema to be RowTypePtr to begin with?

Reply:
@mbasmanova Yes. Updated.